### DISCUSSION TITLE
time_limit and soft_time_limit feature are not working for celery task with SQS

### DISCUSSION THREAD

**Discussion by Knight1997** [celery/celery#8381] [Q&A]
I simply want to timeout the worker if the task is taking too long.

My setup: Running SQS locally using localStack and running 1 worker in solo pool. celery version=5.2.7

I am doing this:
```
upload_task.apply_async(queue=queue, args=(123),  time_limit=5, soft_time_limit=2)

@celery_app.task(name="upload_task")
@async_to_sync
async def upload_task(id: str) -> None:
    try:
        sleep(10)
        logger.info("upload task finished")
    except SoftTimeLimitExceeded:
        logger.exception(f"upload task: {id} failed due to worker timeout")

```

Ideally in this case, it this case, worker should not print string `upload task finished` but it does and the log says: `Task upload_task[2f8b2aed-ad95-44e1-976e-6f078dd0d90] succeeded in 10.0109819220379s: None`

I am expecting It should have been timed out with some timeout exception. but it did not.

**Comment by auvipy** [comment]
I'm not sure about the bug here. are you seeing the intended behavior?

**Reply by Knight1997** [reply]
No, ideally the worker should have stopped working and throw timeout exception and task should have been reassigned to either new/or same worker. I am not seeing any such behavior.

**Reply by proever** [reply]
I'm having the same issue, setting the time limit via `apply_async` seems to have no effect. I have not specified a "default" time limit, and have not experimented with doing so yet.
